### PR TITLE
[Release] Version 2.0.0

### PR DESCRIPTION
**Pull request by Syzuna**
Changelog:
- More or less complete rewrite of the underlying clients to make them more stable and a push for more "asyncness"
- Throttling was removed from Communications and reimplemented in Client
- Reworked logging
- Added AsyncEvents

### CODE REVIEW

File: src/TwitchLib.Communication/Clients/ClientBase.cs
Comment (swiftyspiffy) [inline src/TwitchLib.Communication/Clients/ClientBase.cs:38]: ill assume 15 seconds is a standard practice

File: src/TwitchLib.Communication/TwitchLib.Communication.csproj
Comment (swiftyspiffy) [inline src/TwitchLib.Communication/TwitchLib.Communication.csproj:10]: feel like we should add some reference to Github contributors, since a decent portion of this code is now written by new-ish people :)

**Review by swiftyspiffy — APPROVED** [review]
lgtm

File: src/TwitchLib.Communication/TwitchLib.Communication.csproj
Comment (Syzuna) [reply src/TwitchLib.Communication/TwitchLib.Communication.csproj:10]: I am fine with that.
How would you like to do that?

File: src/TwitchLib.Communication/Clients/ClientBase.cs
Comment (Syzuna) [reply src/TwitchLib.Communication/Clients/ClientBase.cs:38]: Not sure if its a standard practice but its a reasonable amount imo

File: src/TwitchLib.Communication/TwitchLib.Communication.csproj
Comment (swiftyspiffy) [reply src/TwitchLib.Communication/TwitchLib.Communication.csproj:10]: I'm not sure it's realistic to include every new contributor in this list, so maybe just "swiftyspiffy, Prom3theu5, Syzuna, LuckyNoS7evin, and Github contributors" or something? We'd prob need to do something similar for the other packages as well.

Comment (swiftyspiffy) [reply src/TwitchLib.Communication/TwitchLib.Communication.csproj:10]: We can add this later though, lets not block on this.

Comment (Syzuna) [reply src/TwitchLib.Communication/TwitchLib.Communication.csproj:10]: Yeah I would like to add the one other PR to this before merging tho.
Did you have time to look at that one too? Bcs I didnt yet

Comment (swiftyspiffy) [reply src/TwitchLib.Communication/TwitchLib.Communication.csproj:10]: Referring to #27 ? I can look tonight (in a few hours).

**Review by AoshiW — APPROVED** [review]

**Pull request merged**
